test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -8143,6 +8143,28 @@ BOOST_AUTO_TEST_CASE(require_visibility_specifiers)
 	CHECK_ERROR(text, SyntaxError, "No visibility specified.");
 }
 
+BOOST_AUTO_TEST_CASE(blockhash_not_available_in_block)
+{
+	char const* text = R"(
+		contract Test {
+			function a() public returns (bytes32) {
+				return block.blockhash(0);
+			}
+		}
+	)";
+ 	CHECK_ERROR(text, TypeError, "Member \"blockhash\" not found or not visible after argument-dependent lookup in block");
+}
+
+BOOST_AUTO_TEST_CASE(blockhash_local_shadow)
+{
+	char const* text = R"(
+		contract test {
+			function blockhash() pure public returns (bytes32) {}
+		}
```

Comment:
Maybe add another function to the test contract that calls the local blockhash and checks whether the compiler resolves it correctly, e.g. by using a return type that is incompatible to the return type of global blackhash and assigning the result of the function call to a variable?

Reply:
I think it's better to have a function with identical signature and use an EndToEnd test to check.